### DISCUSSION TITLE
How to add task in php language?

### DISCUSSION THREAD

**Discussion by ydhcxh** [hibiken/asynq#464] [Q&A] [answered]
Hi everyone,
     Our project developped by php, How to add task in php language, and consumer in go?   can i  put task into the queue by direct write data into redis db?  I wanted there is a faster way to do this? i need a litter help thanks a lot!

**Comment by hibiken** [comment, marked as answer]
@ydhcxh Thank you for opening a discussion!

Related issue #105, #458.
There're two approaches that we are planning to support in the future. 

1. Provide a client library in different languages

2. Provide a HTTP/gRPC service that applications written in other languages can talk to enqueue tasks 

In the mean time, you'd need to roll this on your own.
If you go with #1 approach, please take a look at the `client.go` for the current implementation. 
Let me know if you have any questions :) Happy to help!

**Reply by ydhcxh** [reply]
Thank you for the answer, Due to time constraints, I temporarily used a compromise solution，First, let PHP put the task into another redis DB, and then asynq's producer takes the task from the queen  enqueue into asynq , the full producer.go source code:

`
package main

import (
	"context"
	"encoding/json"
	"fmt"
	"github.com/go-redis/redis/v8"
	"github.com/hibiken/asynq"
	"log"
	"time"
)
type Tqueue struct {
	// typename indicates the type of task to be performed.
	T string `json:"t"`

	// payload holds data needed to perform the task.
	P string `json:"p"`
    // add to  task times
	C int `json:"c"`
}
var ctx context.Context
var rdb *redis.Client
var asynqClient *asynq.Client
func main() {
	log.Printf("start producer main")
	ctx = context.Background()
	rdb = redis.NewClient(&redis.Options{
		Addr:     "172.16.0.38:6379",
		Password: "", // no password set
		DB:       8,  // use default DB
	})
	ra := asynq.RedisClientOpt{Addr: "172.16.0.38:6379", DB:9}
	asynqClient = asynq.NewClient(ra)

	defer func() {
		rdb.Close()
		asynqClient.Close()
	}()
	for {
		result, err := rdb.BLPop(ctx,10*time.Second,"wdmqueen").Result()

		if err != nil {
			log.Printf(" ... ")
			continue
		}
		log.Printf("get result 0 : %s", result[0])
		log.Printf("get result 1 : %s ", result[1])
		go AddNewTask(result[1])
	}
}
func AddNewTask(data string) error {
	log.Printf("start add new task")
	var t Tqueue
	if err := json.Unmarshal([]byte(data), &t); err != nil {
		log.Printf("Unmarshal err %s,%v",data,err)
		return fmt.Errorf("json.Unmarshal failed: %s: %v", data,err)
	}
	t1 := asynq.NewTask(t.T, []byte(t.P))
	info, err := asynqClient.Enqueue(t1)
	if err != nil {
		log.Fatal(err)
		if t.C <10 {
			t.C +=1
			str, _ := json.Marshal(t)
			rdb.RPush(ctx,"wdmqueen", string(str))
		}
	}
	log.Printf("enqueued task: id=%s queue=%s", info.ID, info.Queue)
	return nil
}

`
